src/main/java/com/ably/kafka/connect/config/ConfigValueEvaluator.java

```diff
@@ -46,13 +46,17 @@ public Result evaluate(SinkRecord record, String pattern, boolean skippable) thr
         if (pattern == null) {
             return new Result(false, null);
         }
-        final byte[] key = (byte[]) record.key();
-        String keyString = null;
 
-        //we only want to evalutate UTF-8 encoded strings
-        if(key != null && ByteArrayUtils.isUTF8Encoded(key)) {
-            keyString = new String(key, StandardCharsets.UTF_8);
+        String keyString = null;
+        final Object recordKey = record.key();
+        if (recordKey != null) {
+            if (recordKey instanceof byte[]  && ByteArrayUtils.isUTF8Encoded((byte[]) recordKey)){
+                keyString = new String((byte[]) recordKey, StandardCharsets.UTF_8);
+            }else if (recordKey instanceof String){
```

Comment:
Spacing around `{` and `}` is inconsistent with other parts of the file - I would suggest we consistently have a space around them